core/src/main/scala/kafka/server/DelayedRemoteFetch.scala

```diff
@@ -80,6 +81,8 @@ class DelayedRemoteFetch(remoteFetchTask: Future[Void],
     // cancel the remote storage read task, if it has not been executed yet
     val cancelled = remoteFetchTask.cancel(true)
     if (!cancelled) debug(s"Remote fetch task for for RemoteStorageFetchInfo: $remoteFetchInfo could not be cancelled and its isDone value is ${remoteFetchTask.isDone}")
+
+    DelayedRemoteFetchMetrics.expiredRequestMeter.mark()
```

Comment:
It would be good if we could log warning when following trying to do remote fetch like I did: https://github.com/apache/kafka/compare/trunk...showuon:KAFKA-16013?expand=1#diff-13a87a3a84b41ec66eb8ce8bc46dc166158f268d214821a3e9eaf28029eceef4R82

Reply:
This makes sense, thanks. I have added the log.